paimon-hive/paimon-hive-connector-common/src/main/java/org/apache/paimon/hive/HiveSchema.java

```diff
@@ -112,21 +111,24 @@ public static HiveSchema extract(@Nullable Configuration configuration, Properti
         List<String> columnNames = Arrays.asList(columnProperty.split(columnNameDelimiter));
         String columnTypes = properties.getProperty(serdeConstants.LIST_COLUMN_TYPES);
         List<TypeInfo> typeInfos = TypeInfoUtils.getTypeInfosFromTypeString(columnTypes);
+
+        // If both Paimon table schema and Hive table schema exist, we check whether the schema
+        // matches and still build from Paimon table schema
+        if (tableSchema.isPresent()) {
+            if (columnNames.size() > 0 && typeInfos.size() > 0) {
+                LOG.debug(
+                        "Extract schema with exists DDL and exists paimon table, table location:[{}].",
+                        location);
+                checkSchemaMatched(columnNames, typeInfos, tableSchema.get());
+            }
+
+            return new HiveSchema(new RowType(tableSchema.get().fields()));
```

Comment:
Don't do this.

For example, Hive DDL can be f1, f2, but table can be f0,f1,f2,f3,f4

Reply:
modified.